dashboard/pages/notifications.py

```diff
@@ -1,6 +1,226 @@
+                Minimum magnitude chosen: {magnitude}""")
+
+        return {'email': email, 'phone_number': phone_number,
+                'selected_lat': selected_lat, 'selected_lon': selected_lon,
+                'magnitude': magnitude.split(" ")[0]}
+    return []
+
+
+def unsubscribe_user_from_topic(client, email_subscription_arn, sms_subscription_arn):
+    """unsubscribes given user from topic that matches the given email and sms subscription arns"""
+    client.unsubscribe(
+        SubscriptionArn=email_subscription_arn
+    )
+    client.unsubscribe(
+        SubscriptionArn=sms_subscription_arn
+    )
+
+
+def get_user_subscription_arn(conn, user_info):
+    """returns a given users email and sms subscription arns"""
+    with conn.cursor as curr:
+        curr.execute(f"""select uta.sms_subscription_arn, uta.email_subscription_arn
+                            from user_topic_assignments as uta
+                            join users as u on u.user_id = uta.user_id
+                            where u.email_address = {user_info['email_address']}
+                            and u.phone_number = {user_info['phone_number']}""")
+        result = curr.fetchone()
+    return result
+
+
+def create_unsubscribe_form():
+    """creates unsubscribe form"""
+    email = ""
+    phone_number = ""
+    with st.form("notification-unsubscribe", clear_on_submit=True, border=True):
+        st.header("Unsubscribe from earthquake notifications")
+        st.write("""If you have previously subscribed to receiving earthquake notifications and wish to unsubscribe,
+                 please enter the email and phone number used when subscribing below:""")
+        email = st.text_input("Email:")
+        phone_number = st.text_input("Phone Number:")
+
+        submit = st.form_submit_button()
+    if submit and email and phone_number:
+        client = get_sns_client()
+        conn = get_connection()
+        user = check_if_user_exists(
+            conn, {'email': email, 'phone_number': phone_number})
+        if user is None:
+            st.error(f'''User with email {email} and phone number {
+                     phone_number} does not exist''')
+        else:
+            sms_subscription_arn, email_subscription_arn = get_user_subscription_arn(
+                conn, {'email_address': email, 'phone_number': phone_number})
+            unsubscribe_user_from_topic(
+                client, email_subscription_arn, sms_subscription_arn)
+
+
+def create_notifications_page():
+    """creates the notification subscription page"""
+    st.set_page_config(
+        page_title="Earthquake Alerts Subscription",)
+    user_info = create_subscription_form()
+    create_unsubscribe_form()
+    if user_info != []:
+        client = get_sns_client()
+        conn = get_connection()
+        if check_if_user_exists(conn, user_info) is None:
+            user_id = upload_user_to_database(conn, user_info)
+        if check_if_topic_exists(conn, user_info) is None:
```

Comment:
I like that you check whether a topic exists before hand, good work considering and working around this case